### DISCUSSION TITLE
GetServerSidePropsContext.req.cookies return undefined when refresh page, but req.headers.cookie has value

### DISCUSSION THREAD

**Discussion by 7-cat** [vercel/next.js#23804] [Help]
### What version of Next.js are you using?

latest

### What version of Node.js are you using?

14.15.4

### What browser are you using?

chrome

### What operating system are you using?

macOS 11.2.3

### How are you deploying your application?

next dev and next start

### Describe the Bug

export const getServerSideProps: GetServerSideProps = async (
  context: GetServerSidePropsContext,
) => {
  const { req, res } = context;
  let token = req.cookies?.token; // req.cookies return undefined when refresh browser page, but req.headers.cookie has value
}

### Expected Behavior

should not be undefined when req.headers.cookie has value

### To Reproduce

export const getServerSideProps: GetServerSideProps = async (
  context: GetServerSidePropsContext,
) => {
  const { req, res } = context;
  let token = req.cookies?.token; // req.cookies return undefined when refresh browser page, but req.headers.cookie has value
}

**Comment by alanyong91** [comment]
How you store the cookies? Can you share your cookies settings such as `httpOnly`, `secure`, `sameSite` and `path`?

**Reply by bnmlynx** [reply]
I am also having a similar issue,  when on the root of the site the cookies is returned but any other page and empty object. Was wondering if you could help if I provide the cookie settings:

cookie settings are 
httpOnly - false 
secure - false
path - /

Wondering if it has something to do with the path?

**Comment by ephraimduncan** [comment]
The cookie is still available in the Request Headers. I was able to access it in the `req.headers` object as below.
```js
const { cookie } = req.headers
```

If you are using the [`cookie`](https://www.npmjs.com/package/cookie) package, you can then parse it as an object.

```js
import  { parse } from "cookie";

const parsedCookie = parse(cookie);
```

Your `getServeSideProps` should look like this
```js
export function getServerSideProps({ req, res }) {
  const { cookie } = req.headers
  const parsedCookie = parse(cookie);

  return {
    props: { cookie: parsedCookie },
  };
}
```
